sample-status-updater-application/src/main/java/life/qbic/samplestatusupdater/app/SampleUpdater.java

```diff
@@ -72,18 +74,30 @@ public void execute() {
         SampleTrackingService sampleTrackingService = new SampleTrackingServiceConnector(serviceList.get(0), credentials);
 
         OpenBisSearchService searchService = new SampleSearchConnector(session.getApi(), session.getToken());
-        Date lastSearchDate = parseDateFromStringWithPattern((String) properties.get("lastSearchDate"), "yyyy-MM-dd");
+        Instant lastSearchDate = null;
+        try{
+            lastSearchDate = Instant.parse((String) properties.get("lastSearchDate"));
+        } catch (Exception e) {
+            LOG.error("Could not parse last search date " + properties.get("lastSearchDate"));
+            Arrays.stream(e.getStackTrace()).forEach( LOG::error );
+            System.exit(1);
+        }
 
+        Instant timePointBeforeSearch = Instant.now();
         SampleUpdatePresenter updatePresenter = new SampleUpdatePresenter();
-
-        final UpdateSampleStatus updateSampleStatus = new UpdateSampleStatusImpl(sampleTrackingService, updatePresenter);
-        UseCaseConnector useCaseConnector = new UseCaseConnector(updateSampleStatus);
-        final FindNewOpenBisSamples findNewOpenBisSamples = new FindNewOpenBisSamples(searchService, useCaseConnector);
-        findNewOpenBisSamples.searchNewSamplesSince(lastSearchDate);
-
-        properties.put("lastSearchDate", new SimpleDateFormat("yyyy-MM-dd").format(new Date()));
+        int exitCode = 0;
+        try {
+            final UpdateSampleStatus updateSampleStatus = new UpdateSampleStatusImpl(sampleTrackingService, updatePresenter);
+            UseCaseConnector useCaseConnector = new UseCaseConnector(updateSampleStatus);
+            final FindNewOpenBisSamples findNewOpenBisSamples = new FindNewOpenBisSamples(searchService, useCaseConnector);
+            findNewOpenBisSamples.searchNewSamplesSince(lastSearchDate);
+        } catch (Exception e) {
+            exitCode = 1;
+        }
```

Comment:
There should be some indication as to what went wrong. Otherwise, the application will exit with `1` and you do not know what happened. Maybe introducing a log message would be a solution?
```suggestion
        } catch (Exception e) {
            LOG.error(e.getMessage());
            exitCode = 1;
        }
```